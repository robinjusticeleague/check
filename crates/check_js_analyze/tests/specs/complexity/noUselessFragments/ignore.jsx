function Component() {
  return (<>
    {/** check-ignore lint/complexity/noUselessFragments: <explanation> */}
    <></>
    {/**     check-ignore lint/complexity/noUselessFragments: <explanation> */}
    <></>
    {/** check-ignore     lint/complexity/noUselessFragments: <explanation> */}
    <></>
    {/** check-ignore lint/complexity/noUselessFragments:       <explanation> */}
    <></>
    {/** check-ignore lint/complexity/noUselessFragments: <explanation>     */}
    <></>
  </>)
}