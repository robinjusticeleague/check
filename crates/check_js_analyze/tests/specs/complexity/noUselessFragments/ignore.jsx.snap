---
source: crates/check_js_analyze/tests/spec_tests.rs
expression: ignore.jsx
---
# Input
```jsx
function Component() {
  return (<>
    {/** check-ignore lint/complexity/noUselessFragments: <explanation> */}
    <></>
    {/**     check-ignore lint/complexity/noUselessFragments: <explanation> */}
    <></>
    {/** check-ignore     lint/complexity/noUselessFragments: <explanation> */}
    <></>
    {/** check-ignore lint/complexity/noUselessFragments:       <explanation> */}
    <></>
    {/** check-ignore lint/complexity/noUselessFragments: <explanation>     */}
    <></>
  </>)
}
```

# Diagnostics
```
ignore.jsx:3:59 suppressions/incorrect ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! A suppression shouldn't have an <explanation> placeholder. Example of suppression: // check-ignore lint: false positive
  
    1 │ function Component() {
    2 │   return (<>
  > 3 │     {/** check-ignore lint/complexity/noUselessFragments: <explanation> */}
      │                                                           ^^^^^^^^^^^^^
    4 │     <></>
    5 │     {/**     check-ignore lint/complexity/noUselessFragments: <explanation> */}
  

```

```
ignore.jsx:5:63 suppressions/incorrect ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! A suppression shouldn't have an <explanation> placeholder. Example of suppression: // check-ignore lint: false positive
  
    3 │     {/** check-ignore lint/complexity/noUselessFragments: <explanation> */}
    4 │     <></>
  > 5 │     {/**     check-ignore lint/complexity/noUselessFragments: <explanation> */}
      │                                                               ^^^^^^^^^^^^^
    6 │     <></>
    7 │     {/** check-ignore     lint/complexity/noUselessFragments: <explanation> */}
  

```

```
ignore.jsx:7:63 suppressions/incorrect ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! A suppression shouldn't have an <explanation> placeholder. Example of suppression: // check-ignore lint: false positive
  
    5 │     {/**     check-ignore lint/complexity/noUselessFragments: <explanation> */}
    6 │     <></>
  > 7 │     {/** check-ignore     lint/complexity/noUselessFragments: <explanation> */}
      │                                                               ^^^^^^^^^^^^^
    8 │     <></>
    9 │     {/** check-ignore lint/complexity/noUselessFragments:       <explanation> */}
  

```

```
ignore.jsx:9:65 suppressions/incorrect ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! A suppression shouldn't have an <explanation> placeholder. Example of suppression: // check-ignore lint: false positive
  
     7 │     {/** check-ignore     lint/complexity/noUselessFragments: <explanation> */}
     8 │     <></>
   > 9 │     {/** check-ignore lint/complexity/noUselessFragments:       <explanation> */}
       │                                                                 ^^^^^^^^^^^^^
    10 │     <></>
    11 │     {/** check-ignore lint/complexity/noUselessFragments: <explanation>     */}
  

```

```
ignore.jsx:11:59 suppressions/incorrect ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! A suppression shouldn't have an <explanation> placeholder. Example of suppression: // check-ignore lint: false positive
  
     9 │     {/** check-ignore lint/complexity/noUselessFragments:       <explanation> */}
    10 │     <></>
  > 11 │     {/** check-ignore lint/complexity/noUselessFragments: <explanation>     */}
       │                                                           ^^^^^^^^^^^^^
    12 │     <></>
    13 │   </>)
  

```
